### PR TITLE
#604: Fix MAVEN_ARGS Path Format for Windows

### PR DESCRIPTION
**Pull request by alfeilex**
Fixes: #604 

Tested by creating a native image for Windows

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/context/IdeContext.java [hunk 1]
```diff
@@ -438,11 +438,11 @@ default String getMavenArgs() {
       return null;
     }
     String settingsPath;
-    WindowsPathSyntax pathSyntax = getPathSyntax();
-    if (pathSyntax == null) {
+    boolean isWindows = getSystemInfo().isWindows();
```
Comment (jan-vcapgemini) [inline cli/src/main/java/com/devonfw/tools/ide/context/IdeContext.java:441]: Do we need this boolean somewhere else too? If not, we can replace it in the if condition.

**Review by jan-vcapgemini — CHANGES_REQUESTED** [review]
LGTM. The isWindows boolean can be removed I think.

**Review by hohwille — APPROVED** [review]
@alfeilex thanks for creating the bug issue and fixing PR. 👍 
I left a comment as you can simplify this to a single line since OS check is pointless. You do not need `WindowsPathSyntax` to convert from WINDOWS to WINDOWS since `Path` on Windows will always format in Windows syntax via `toString()`.

BTW: This is now resulting in the code how we wrote it before it was broken with #321

File: cli/src/main/java/com/devonfw/tools/ide/context/IdeContext.java (hunk 1 above)
Comment (alfeilex) [reply cli/src/main/java/com/devonfw/tools/ide/context/IdeContext.java]: Has been done with the removal of the OS checking

File: cli/src/main/java/com/devonfw/tools/ide/context/IdeContext.java [hunk 2]
```diff
@@ -438,11 +438,11 @@ default String getMavenArgs() {
...
+    if (!isWindows) {
       settingsPath = mvnSettingsFile.toString();
     } else {
-      settingsPath = pathSyntax.format(mvnSettingsFile);
+      settingsPath = WindowsPathSyntax.WINDOWS.format(mvnSettingsFile);
     }
```
[the remark this answers is not in the document]
Comment (alfeilex) [reply cli/src/main/java/com/devonfw/tools/ide/context/IdeContext.java]: Done

**Review by jan-vcapgemini — APPROVED** [review]
LGTM.

**Pull request merged**